localstack/http/dispatcher.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any, Dict, Protocol, Union
+
+from werkzeug.exceptions import MethodNotAllowed
+
+from .request import Request
+from .response import Response
+from .router import Dispatcher, RequestArguments
+
+ResultValue = Union[
+    Response,
+    str,
+    bytes,
+    Dict[str, Any],  # a JSON dict
+]
+
+
+def _populate_response(response: Response, result: ResultValue):
+    if isinstance(result, dict):
+        response.set_json(result)
+    else:
+        response.data = result
```

Comment:
yep good catch! i think adding the check here would be incoherent, since we're already passing a response object to populate. but i added the checks in the calling methods, which makes total sense.